src/Templates/DotnetCore/Microsoft.NetCore.VB.ProjectTemplates.Desktop/Winforms.vstemplate

```diff
@@ -0,0 +1,29 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<VSTemplate Version="3.0.0" Type="ProjectGroup" xmlns="http://schemas.microsoft.com/developer/vstemplate/2005" xmlns:sdk="http://schemas.microsoft.com/developer/vstemplate-sdkextension/2010">
+    <TemplateData>
+        <Name>Windows Forms App (.NET Core)</Name>
+        <Description Package="{164B10B9-B200-11D0-8C61-00A0C91E29D5}" ID="3015" />
+        <Icon Package="{164B10B9-B200-11D0-8C61-00A0C91E29D5}" ID="4507" />
+        <ProjectType>VisualBasic</ProjectType>
+        <SortOrder>12</SortOrder>
+        <NumberOfParentCategoriesToRollUp>1</NumberOfParentCategoriesToRollUp>
+        <TemplateID>Microsoft.VB.NETCore.Winforms</TemplateID>
+        <CreateNewFolder>true</CreateNewFolder>
+        <CreateInPlace>true</CreateInPlace>
+        <DefaultName>WindowsFormsApp</DefaultName>
+        <ProvideDefaultName>true</ProvideDefaultName>
+        <PromptForSaveOnCreation>true</PromptForSaveOnCreation>
+    </TemplateData>
+    <TemplateContent>
+        <ProjectCollection/>
+        <CustomParameters>
+            <CustomParameter Name="$language$" Value="VB" />
+            <CustomParameter Name="$uistyle$" Value="none" />
+            <CustomParameter Name="$groupid$" Value="Microsoft.Common.WinForms" />
```

Comment:
add
```xml
<CustomParameter Name="$Framework$" Value="net5.0" />
```

Reply:
Thanks!

Reply:
after digging through in the debugger I think these are the values you want:
```xml
<CustomParameter Name="$targetframeworkversion$" Value="net5.0" />
<CustomParameter Name="$passthrough:useappframework$" Value="true" />
```

The strings that the template wizard uses to filter the custom parameters are defined [here](https://devdiv.visualstudio.com/DevDiv/_git/WebTools?path=/src/TemplateEngine/Microsoft.VisualStudio.TemplateEngine.Wizard/WizardConstants.cs)

The code that filters out the parameters and adds them to the dictionary that is given to the template engine is [here](https://devdiv.visualstudio.com/DevDiv/_git/WebTools?path=%2Fsrc%2FTemplateEngine%2FMicrosoft.VisualStudio.TemplateEngine.Wizard%2FTemplateWizardPipelineBase.cs&version=GBmain&line=92&lineEnd=106&lineStartColumn=1&lineEndColumn=14&lineStyle=plain)

This dictionary is then passed to the CLI template engine [here](https://github.com/dotnet/templating/blob/bfaba139963bd958088e2ac1069a336b1fa3721d/src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs#L58) in the `inputParameters` dictionary.

Reply:
So, after debugging through the template engine, I found out it's done like this:
```
            <CustomParameter Name="$platformversion$" Value="net5.0" />
```

The `UseAppFramework` we no longer need, since we covered the differentiation between CLI and VS in the CLI template:
https://github.com/dotnet/winforms/pull/3967